src/buffer/out/textBuffer.hpp

```diff
@@ -330,16 +295,17 @@ class TextBuffer final
     std::vector<til::point_span> SearchText(const std::wstring_view& needle, bool caseInsensitive) const;
     std::vector<til::point_span> SearchText(const std::wstring_view& needle, bool caseInsensitive, til::CoordType rowBeg, til::CoordType rowEnd) const;
 
-    const std::vector<ScrollMark>& GetMarks() const noexcept;
+    // Mark handling
+    std::vector<ScrollMark> GetMarkRows() const;
+    std::vector<MarkExtents> GetMarkExtents(std::optional<size_t> limit = std::nullopt) const;
     void ClearMarksInRange(const til::point start, const til::point end);
-    void ClearAllMarks() noexcept;
-    void ScrollMarks(const int delta);
-    void StartPromptMark(const ScrollMark& m);
-    void AddMark(const ScrollMark& m);
-    void SetCurrentPromptEnd(const til::point pos) noexcept;
-    void SetCurrentCommandEnd(const til::point pos) noexcept;
-    void SetCurrentOutputEnd(const til::point pos, ::MarkCategory category) noexcept;
-    std::wstring_view CurrentCommand() const;
+    void ClearAllMarks();

```

Comment:
before I get into the implementation-q: this sounds like it will clear all scrollbar marks _and_ attribute marks. Is that so?

Reply:
(it did)